source/GameInterface/DynamicSync/DynamicSyncBuilder.cs

```diff
@@ -0,0 +1,64 @@
+﻿using GameInterface.DynamicSync;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace GameInterface.AutoSync;
+public interface IDynamicSyncBuilder : IDisposable
+{
+    /// <summary>
+    /// Add a field to automatically sync across the network
+    /// </summary>
+    /// <param name="field">Field to auto sync</param>
+    void AddField(FieldInfo field);
+
+    /// <summary>
+    /// Add a property to automatically sync across the network
+    /// </summary>
+    /// <param name="property">Property to auto sync</param>
+    void AddProperty(PropertyInfo property);
+
+    /// <summary>
+    /// Add further TargetMethods that access the sync properties/fields
+    /// </summary>
+    /// <param name="type">Type of the TargetClass the properties/fields belong to</param>
+    /// <param name="methodInfo">MethodInfo of method outside TargetClass</param>
+    void AddTargetMethod(Type type, MethodInfo methodInfo);
```

Comment:
Type is not needed, use MethodInfo.Declaring type to get the type
```suggestion
    void AddTargetMethod(MethodInfo methodInfo);
```

Reply:
This is wrong as the type in front is used for the type that is getting patches to collect all TargetMethods where it gets referenced. For e.g. if we collect the info for town we want to add BuildingHelper.ChangeCurrentBuildingQueue as a TargetMethod for the Town Patches